llvm/include/llvm/Support/DXILConstants.def

```diff
@@ -0,0 +1,76 @@
+
+#ifdef DXIL_MODULE_FLAG
+
+// DXIL_MODULE_FLAG(bit offset for the flag, bit offset for feature flag (-1 for no feature flag), name, description.
+DXIL_MODULE_FLAG( 0, -1,  DisableOptimizations,   "D3D11_1_SB_GLOBAL_FLAG_SKIP_OPTIMIZATION")
+DXIL_MODULE_FLAG( 1, -1,  DisableMathRefactoring, "D3D10_SB_GLOBAL_FLAG_REFACTORING_ALLOWED")
+DXIL_MODULE_FLAG( 2,  0,  EnableDoublePrecision,  "Double-precision floating point")
```

Comment:
A bunch of these flags are duplicated in the DXContainerCosntants. Is there a reason we need to define them twice?

Reply:
We could share, just need to add a field to SHADER_FEATURE_FLAG for DXILModuleFlag bit offset.
If we put all things as SHADER_FEATURE_FLAG, there will be something not SHADER_FEATURE_FLAG though.